sklearn/metrics/pairwise.py

```diff
@@ -1305,6 +1305,13 @@ def sigmoid_kernel(X, Y=None, gamma=None, coef0=1):
     return K
 
 
+@validate_params(
+    {
+        "X": ["array-like", "sparse matrix"],
+        "Y": ["array-like", "sparse matrix", None],
+        "gamma": [Interval(Real, None, None, closed="neither"), None],
```

Comment:
Also thinking that only positive real makes sense here since this is the width of the neighborhood when applying the kernel.

Reply:
As for others, it is used in the Gaussian Process. So we need an Hiden option for ndarray.